hypothesis-python/src/hypothesis/ghostwriter.py

```diff
@@ -86,16 +122,22 @@ def _strategy_for(param: inspect.Parameter) -> Union[st.SearchStrategy, InferTyp
     return st.just(param.default)
 
 
-def _get_params(func: Callable) -> Dict[str, inspect.Parameter]:
+def _get_params(
+    func: Callable, skip_first: bool = False
+) -> Dict[str, inspect.Parameter]:
     var_param_kinds = (inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD)
     params = list(inspect.signature(func).parameters.values())
+    if skip_first:
+        del params[0]
     return OrderedDict((p.name, p) for p in params if p.kind not in var_param_kinds)
 
 
-def _get_strategies(*funcs: Callable) -> Dict[str, st.SearchStrategy]:
+def _get_strategies(
+    *funcs: Callable, passon: bool = False
+) -> Dict[str, st.SearchStrategy]:
```

Comment:
I can't really tell what this function is doing. In particular I don't know what the `passon` parameter is for. Could it get a docstring?